crates/wysiwyg/src/composer_model/format.rs

```diff
@@ -190,6 +267,98 @@ where
+            if let DomNode::Container(node) = node {
+                match node.kind() {
+                    ContainerNodeKind::Formatting(f) => {
+                        if f.clone() == format.clone() {
+                            let index_in_parent =
+                                location.node_handle.index_in_parent();
+                            let parent_node = self.state.dom.lookup_node_mut(
+                                &location.node_handle.parent_handle(),
+                            );
+                            if let DomNode::Container(parent_node) = parent_node
+                            {
+                                parent_node.remove_child_and_own_hierarchy(
+                                    index_in_parent,
+                                );
+                            }
+
+                            // Re-apply formatting to slices before and after the selection if needed
+                            let (before_start, before_end) = self
+                                .safe_locations_from(
+                                    Location::from(
+                                        start - location.start_offset,
+                                    ),
+                                    Location::from(start),
+                                );
+                            let (after_start, after_end) = self
+                                .safe_locations_from(
+                                    Location::from(end),
+                                    Location::from(
+                                        end + location.length
+                                            - location.end_offset,
+                                    ),
+                                );
+
+                            if before_end > before_start {
+                                self.format_range(
+                                    before_start,
+                                    before_end,
+                                    format.clone(),
+                                );
+                            }
+                            if after_end > after_start {
+                                self.format_range(
+                                    after_start,
+                                    after_end,
+                                    format.clone(),
+                                );
+                            }
+                        }
+                    }
+                    _ => {}
+                }
+            }
+        }
+    }
+
+    fn find_parent_formatting_node(
+        &self,
+        handle: DomHandle,
+        format: InlineFormatType,
+    ) -> DomHandle {
+        let node = self.state.dom.lookup_node(&handle);
+        if let DomNode::Container(container) = node {
+            match container.kind() {
+                ContainerNodeKind::Formatting(f) => {
+                    if f.clone() == format {
```

Comment:
Same here, `*f` is probably enough to use the reference's contents.

```suggestion
                    if *f == format {
```